packages/core/admin/admin/src/content-manager/pages/App/LeftMenu/index.js

```diff
@@ -4,54 +4,106 @@
  *
  */
 
-import React, { useMemo } from 'react';
-import { LeftMenuList, colors, sizes } from '@strapi/helper-plugin';
-import styled from 'styled-components';
+import React, { useMemo, useState } from 'react';
+import {
+  SubNav,
+  SubNavHeader,
+  SubNavSection,
+  SubNavSections,
+  SubNavLink,
+} from '@strapi/parts/SubNav';
 import { useSelector, shallowEqual } from 'react-redux';
+import { useIntl } from 'react-intl';
+import matchSorter from 'match-sorter';
+import sortBy from 'lodash/sortBy';
+import toLower from 'lodash/toLower';
 import getTrad from '../../../utils/getTrad';
 import { makeSelectModelLinks } from '../selectors';
 
-const Wrapper = styled.div`
-  width: 100%;
-  min-height: calc(100vh - ${sizes.header.height});
-  background-color: ${colors.leftMenu.mediumGrey};
-  padding-top: 3.1rem;
-  padding-left: 2rem;
-  padding-right: 2rem;
-`;
+const matchByTitle = (links, search) =>
+  matchSorter(links, toLower(search), { keys: [item => toLower(item.title)] });
 
 const LeftMenu = () => {
+  const [search, setSearch] = useState('');
+  const { formatMessage } = useIntl();
   const modelLinksSelector = useMemo(makeSelectModelLinks, []);
   const { collectionTypeLinks, singleTypeLinks } = useSelector(
     state => modelLinksSelector(state),
     shallowEqual
   );
 
-  const data = [
+  const menu = [
     {
-      name: 'models',
+      id: 'collectionTypes',
       title: {
-        id: getTrad('components.LeftMenu.collection-types.'),
+        id: getTrad('components.LeftMenu.collection-types'),
+        defaultMessage:
+          '{number, plural, =0 {Collection Types} one {Collection Type } other {Collection Types}}',
+        values: { number: collectionTypeLinks.length },
       },
       searchable: true,
-      links: collectionTypeLinks,
+      links: sortBy(matchByTitle(collectionTypeLinks, search), 'title'),
     },
     {
-      name: 'singleTypes',
+      id: 'singleTypes',
       title: {
-        id: getTrad('components.LeftMenu.single-types.'),
+        id: getTrad('components.LeftMenu.single-types'),
+        defaultMessage:
+          '{number, plural, =0 {Single Types} one {Single Type } other {Single Types}}',
+        values: { number: singleTypeLinks.length },
       },
       searchable: true,
-      links: singleTypeLinks,
+      links: sortBy(matchByTitle(singleTypeLinks, search), 'title'),
```

Comment:
For the CTB I kept the old search behaviour: Sections with no results are hidden.
Which behaviour should we enforce ?

Reply:
Honestly I don't know